### PR TITLE
Feature item patch support

### PR DESCRIPTION
**Pull request by Obus**
Add `patch` & `patch_batch` methods, other items' methods refactored
See `patch` & `patch_batch` in ingestion PR: https://github.com/Constructor-io/autocomplete/pull/1088

### CODE REVIEW

**Comment by Obus** [comment]
[[ch3514]](https://app.clubhouse.io/constructorio/story/3514/as-an-engineer-i-would-like-to-have-a-patch-like-complex-types-compliant-ingestion-endpoint-so-that-we-can-insert-weighted)

File: constructor_io/constructor_io.py [hunk 1]
```diff
@@ -262,3 +214,33 @@ def track_search(self, term, **kwargs):
...
+            method_name=method_name, method_url="v1/batch_items", request_method=request_method,
+            data=data, url_params=url_params
+        )
+
+    def _format_items_batch_data(self, items, autocomplete_section, **kwargs):
+        for item in items:
+            self._add_params_from_kwargs(item, **kwargs)
+        return {"items": items, "autocomplete_section": autocomplete_section}
+
+    def _non_retrieve_method(self, method_name, method_url, request_method, data,
```
Comment (jcollado) [inline constructor_io/constructor_io.py]: What would be a non retrieve method? A method that is not `GET`?

Comment (Obus) [reply constructor_io/constructor_io.py]: added docstring in https://github.com/Constructor-io/constructorio-python/pull/9/commits/d3315a56689024b7ce6e60ceeb97b2d15f1a7a9c

File: fixtures/ac.cnstrc.com/patch-success.yaml [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+      connection: [keep-alive]
+      content-length: ['0']
+      content-type: [text/html; charset=utf-8]
+      date: ['Thu, 13 Oct 2016 08:30:47 GMT']
+      server: [nginx/1.4.6 (Ubuntu)]
+      strict-transport-security: [max-age=31536000]
+      via: [1.1 varnish]
+      x-varnish: ['1124804859']
+    status: {code: 204, message: NO CONTENT}
+- request:
```
[the remark this answers is not in the document]
Comment (Obus) [reply fixtures/ac.cnstrc.com/patch-success.yaml]: good catch! removed in https://github.com/Constructor-io/constructorio-python/pull/9/commits/71e5dd6fe6061b58b6af522d79703bbee4a26103

**Pull request merged**